jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/annotations/AnnotationDiagnosticsCollector.java

```diff
@@ -192,6 +197,7 @@ public void collectDiagnostics(ICompilationUnit unit, List<Diagnostic> diagnosti
                                         "A method with the annotation @PostConstruct must not throw checked exceptions.");
                                 diagnostic.setCode(AnnotationConstants.DIAGNOSTIC_CODE_POSTCONSTRUCT_EXCEPTION);
                                 diagnostic.setSeverity(DiagnosticSeverity.Warning);
+                                completeDiagnostic(diagnostic);
```

Comment:
this is overwriting the Diagnostic severity from warning to error. Either change the order so that it is 
```java
completeDiagnostic(diagnostic);
diagnostic.setSeverity(DiagnosticSeverity.Warning);
```

or add the source to this diagnostic:
```java
diagnostic.setSeverity(DiagnosticSeverity.Warning);
diagnostic.setSource(AnnotationConstants.DIAGNOSTIC_SOURCE);
```